### PR TITLE
Clarify that direct_post endpoint response is JSON.

### LINKED ISSUE #272: Further clarify `redirect_uri` shape with `direct_post` responses
During a hackathon around UC5 Potential we came across interoperability issues due to a slight misinterpretation of [Section 6.2](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-6.2-22), particularly:

> The following new parameter is defined for use in the response from the endpoint: redirect_uri

combined with

> The following is a non-normative example of the response from the Verifier to the Wallet upon receiving the Authorization Response at the Response Endpoint (using a response_code parameter from [Section 11.5](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#implementation_considerations_direct_post)):
> 
> HTTP/1.1 200 OK
> Content-Type: application/json
> Cache-Control: no-store
> 
> {
>   "redirect_uri": "https://client.example.org/cb#response_code=091535f699ea575c7937fa5f0f454aee"
> }`

Other than the "non-normative example", it was never clearly stated that the `redirect_uri` will return in the response body as a JSON object and there were interpretations that used location headers instead. 

I think stating that the response body is supposed to be a JSON object with just 1 field (`redirect_uri`) is enough to make it 100% clear.

### PR DESCRIPTION
**Pull request by jogu**
For completeness also clarify that unknown parameters must be ignored as we've recently done elsewhere.

closes #272

### CODE REVIEW

**Review by Sakurann — APPROVED** [review]

File: openid-4-verifiable-presentations-1_0.md [hunk 1]
```diff
@@ -638,7 +638,7 @@ It has been defined to address the following use cases:
...
 
 `direct_post`:
-: In this mode, the Authorization Response is sent to the Verifier using an HTTP POST request to an endpoint controlled by the Verifier. The Authorization Response parameters are encoded in the body using the `application/x-www-form-urlencoded` content type. The names and values in the body MUST be encoded using UTF-8. The flow can end with an HTTP POST request from the Wallet to the Verifier, or it can end with a redirect that follows the HTTP POST request, if the Verifier responds with a redirect URI to the Wallet.
+: In this mode, the Authorization Response is sent to the Verifier using an HTTP POST request to an endpoint controlled by the Verifier. The Authorization Response parameters are encoded in the body using the `application/x-www-form-urlencoded` content type. The names and values in the body MUST be encoded using UTF-8. The response from the Verifier is a JSON object as defined below. The flow can end with an HTTP POST request from the Wallet to the Verifier, or it can end with a redirect that follows the HTTP POST request, if the Verifier responds with a redirect URI to the Wallet.
```
Comment (tplooker) [inline openid-4-verifiable-presentations-1_0.md:641]:
```suggestion
: In this mode, the Authorization Response is sent to the Verifier using an HTTP POST request to an endpoint controlled by the Verifier. In this mode the Authorization Response MUST be encoded in the request body using the format defined by the `application/x-www-form-urlencoded` HTTP content type. The parameters in the request body MUST all be encoded using UTF-8. The body of the HTTP response to this request MUST be a JSON object as defined below and designated with the HTTP header of `application/json`. The JSON object in the HTTP response body MAY include a `redirect_uri` parameter which is to be executed as a redirect by the wallet to complete the flow.
```

Tried to make this a little clearer, not precious about the changes though.

File: openid-4-verifiable-presentations-1_0.md [hunk 2]
```diff
@@ -699,13 +699,15 @@ Content-Type: application/x-www-form-urlencoded
   state=eyJhb...6-sVA
 ```
 
-If the Response URI has successfully processed the Authorization Response or Authorization Error Response, it MUST respond with HTTP status code 200.
+If the Response URI has successfully processed the Authorization Response or Authorization Error Response, it MUST respond with HTTP status code 200 and a JSON object.
```
Comment (tplooker) [inline openid-4-verifiable-presentations-1_0.md:702]:
```suggestion
If the Response URI has successfully processed the Authorization Response or Authorization Error Response, it MUST respond with HTTP status code of 200 and a JSON object in the response body.
```

File: openid-4-verifiable-presentations-1_0.md (hunk 1 above)
Comment (jogu) [reply openid-4-verifiable-presentations-1_0.md:641]: Refining this a little as I don't want to be normatively defining the response in 2 places:

```suggestion
: In this mode, the Authorization Response is sent to the Verifier using an HTTP POST request to an endpoint controlled by the Verifier. The Authorization Response MUST be encoded in the request body using the format defined by the `application/x-www-form-urlencoded` HTTP content type. The parameters in the request body MUST all be encoded using UTF-8. The verifier can request that the wallet redirects the user to the verifier using the response as defined below.
```

File: openid-4-verifiable-presentations-1_0.md (hunk 2 above)
Comment (jogu) [reply openid-4-verifiable-presentations-1_0.md:702]: Refined to include the text you'd suggested above:

```suggestion
If the Response URI has successfully processed the Authorization Response or Authorization Error Response, it MUST respond with an HTTP status code of 200 with `Content-Type` of `application/json` and a JSON object in the response body.
```

**Review by tplooker — APPROVED** [review]
LGTM now thanks

**Review by awoie — APPROVED** [review]

**Comment by jogu** [comment]
Editorial & 4 approvals (thank you!) - merging!

**Pull request merged**